libs/wingc/src/parser.rs

```diff
@@ -332,6 +332,43 @@ impl<'s> Parser<'s> {
 		})
 	}
 
+	fn build_struct_definition_statement(&self, statement_node: &Node) -> DiagnosticResult<StmtKind> {
+		let name = self.node_symbol(
+			&statement_node
+				.child_by_field_name("name")
+				.expect("struct definitions should always have a name"),
+		)?;
+
+		let mut cursor = statement_node.walk();
+		let mut members = vec![];
+
+		for field_node in statement_node.children_by_field_name("field", &mut cursor) {
+			let identifier = self.node_symbol(
+				&field_node
+					.child_by_field_name("name")
+					.expect("struct definition fields should always have a name"),
+			)?;
+			let type_ = field_node
+				.child_by_field_name("type")
+				.expect("struct definition fields should always have explicit type");
+			let f = ClassField {
+				name: identifier,
+				member_type: self.build_type_annotation(&type_)?,
+				reassignable: false,
+				phase: Phase::Preflight,
```

Comment:
Is it preflight? I feel like the fields themselves may be phase independent, not really sure though.